PhotoFeed/app/src/main/java/com/mobileproto/david/photofeed/FeedFragment.java

```diff
@@ -0,0 +1,159 @@
+package com.mobileproto.david.photofeed;
+
+
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+
+import java.util.ArrayList;
+
+/**
+ * Created by david on 10/2/15. This class extends fragment, and is the representation of the user's
+ * actual photo feed. It contains functionality to remove photos from the DB.
+ */
+public class FeedFragment extends Fragment
+{
+
+    private static final String DEBUG_TAG = "feedFragmentDebug";
+    private static final String ERROR_TAG = "feedFragmentError";
+    private static final String POS_KEY = "pos";
+
+    private Button back, next;
+    private ImageView imgView;
+    private int pos;
+
+    // The current image url being displayed
+    private String currUrl;
+
+    // The back button's listener. Decrement the current position and update the current photo
+    private View.OnClickListener backListener = new View.OnClickListener()
+    {
+        @Override
+        public void onClick(View v)
+        {
+            pos -= 1;
+            updateFeed(((MainActivity) getActivity()).getmDbHelper());
+        }
+    };
+
+    // The forward button's listener. Decrement the current position and update the current photo
+    private View.OnClickListener fwdListener = new View.OnClickListener()
+    {
+        @Override
+        public void onClick(View v)
+        {
+            pos += 1;
+            updateFeed(((MainActivity) getActivity()).getmDbHelper());
+        }
+    };
+
+    // The long click listener on the image view. When clicked, bring up a dialog and prompts the
+    // user to confirm to the image removal from the DP. Then, updates the image.
+    private View.OnLongClickListener longListener = new View.OnLongClickListener()
+    {
+        @Override
+        public boolean onLongClick(View v)
+        {
+            AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+            builder.setMessage("Remove image from feed?");
+            builder.setPositiveButton("Yes", new DialogInterface.OnClickListener()
+            {
+                @Override
+                public void onClick(DialogInterface dialog, int which)
+                {
+                    ((MainActivity) getActivity()).removeUrlFromDb(currUrl);
+                    updateFeed(((MainActivity) getActivity()).getmDbHelper());
+                    dialog.cancel();
+                }
+            });
+            builder.setNegativeButton("Cancel", null);
+            builder.show();
+            return false;
+        }
+    };
+
```

Comment:
using butterknife for the above Listeners would save you a lot of space, however, either way is perfectly fine!